internal/server/routes.go

```diff
@@ -95,24 +96,30 @@ func (s *Server) GenerateRoutes(promRegistry prometheus.Registerer) Routes {
 	post(a, authn, "/api/tokens", a.CreateToken)
 	post(a, authn, "/api/logout", a.Logout)
 
+	put(a, authn, "/api/settings", a.UpdateSettings)
+
 	authn.GET("/api/debug/pprof/*profile", pprofHandler)
 
-	// these endpoints do not require authentication
-	noAuthn := apiGroup.Group("/", unauthenticatedMiddleware(a.server))
-	post(a, noAuthn, "/api/signup", a.Signup)
+	// no auth required, org not required
+	noAuthnNoOrg := apiGroup.Group("/", unauthenticatedMiddleware(a.server))
+	post(a, noAuthnNoOrg, "/api/signup", a.Signup)
+	get(a, noAuthnNoOrg, "/api/version", a.Version)
 
-	post(a, noAuthn, "/api/login", a.Login)
-	post(a, noAuthn, "/api/password-reset-request", a.RequestPasswordReset)
-	post(a, noAuthn, "/api/password-reset", a.VerifiedPasswordReset)
+	// while an org is required for these endpoints, they return fake data when the org is not supplied
+	noAuthnOptOrg := apiGroup.Group("/", setOrganizationInCtx(a.server), unauthenticatedMiddleware(a.server))
+	get(a, noAuthnOptOrg, "/api/providers", a.ListProviders)
+	get(a, noAuthnOptOrg, "/api/settings", a.GetSettings)
 
-	get(a, noAuthn, "/api/providers", a.ListProviders)
-	get(a, noAuthn, "/api/providers/:id", a.GetProvider)
+	// no auth required, org required
+	noAuthnWithOrg := apiGroup.Group("/", setOrganizationInCtx(a.server), unauthenticatedMiddleware(a.server), orgRequired())
 
-	get(a, noAuthn, "/api/version", a.Version)
+	post(a, noAuthnWithOrg, "/api/login", a.Login)
+	post(a, noAuthnWithOrg, "/api/password-reset-request", a.RequestPasswordReset)
+	post(a, noAuthnWithOrg, "/api/password-reset", a.VerifiedPasswordReset)
 
-	get(a, noAuthn, "/api/settings", a.GetSettings)
-	put(a, authn, "/api/settings", a.UpdateSettings)
-	add(a, noAuthn, route[api.EmptyRequest, WellKnownJWKResponse]{
+	get(a, noAuthnWithOrg, "/api/providers/:id", a.GetProvider)
```

Comment:
There's probably no way for someone to guess a provider ID, but would we want to also return fake data here?

Reply:
we might need to. Not entirely sure yet. provider IDs are somewhat guessable if you have an unlimited number of guesses. Also I think there might be problems with generating reliably-convincing fake data. especially since we're open source.